source/common/http/header_map_impl.h

```diff
@@ -54,9 +54,17 @@ class HeaderMapImpl : public HeaderMap {
   bool operator==(const HeaderMapImpl& rhs) const;
 
   // Http::HeaderMap
+  // Add without copying either key or value.
```

Comment:
nit: Please remove the added comments here. In general we don't replicate comments for functions which are already heavily commented in this interface.